### PR TITLE
Add osx agent, OSX-compatible mk-job

### PR DESCRIPTION
**Pull request by catchdave**
**Better OSX agent:**
* I adapted the OSX version of the agent originally found at: [ThomasKaiser/Check_MK](https://github.com/ThomasKaiser/Check_MK). This seems to be more modern that the version in this repo, I believe. 
* I added OSX support for` lnx_if` since the original `netctr` is no longer supported in check_mk.
* I fixed a bug that breaks smartctl stats for NVME drives in OSX
* Added a OSX version for mk-job (`<<<job>>>`). Different than original since `head` is non-GNU

**Modified mk-job for OSX**
* OSX is BSD-based and therefore `time` does not work as written. It is common for OSX systems to use brew to install GNU-tools, so mk-job will now work as long as `gnu-time` is installed.

**Installer for OSX agent**
I could not find instructions for installing the OSX agent anywhere. So I put together a simple script that I used to install on my mac machines. I have done ZERO testing outside of my 2 machines.

### CODE REVIEW

**Comment by jonaskluger** [comment]
Hello catchdave,

thank you for this pull request. We will have a closer look at your patch and discuss internally on how and if we can implement it into the official code base.

Internal Ref: CMK-4103

**Comment by Notausstieg0309** [comment]
I have tested your proposed MacOS Agent on my MacMini's with macOS 10.15.4. 

Regarding the `lnx_if`: Your implementation does not work for my machines as they have no homebrew software installed. So I have used the implementation from #180 together with your agent, as it supports plugins which is essential to me.

File: agents/check_mk_agent.macosx [hunk 1]
```diff
@@ -1,44 +1,172 @@
...
+
+# Remove locale settings to eliminate localized outputs where possible
 export LC_ALL=C
 unset LANG
 
-export MK_LIBDIR="/to/be/changed"
-export MK_CONFDIR="/to/be/changed"
+export MK_LIBDIR='/usr/local/lib/check_mk_agent'
+export MK_CONFDIR='/etc/check_mk'
+export MK_VARDIR='/var/lib/check_mk_agent'
```
Comment (Notausstieg0309) [inline agents/check_mk_agent.macosx:18]: Please honor the environment before using the default paths as the official linux agent does as well:
```suggestion
export MK_LIBDIR=${MK_LIBDIR:-/usr/local/lib/check_mk_agent}
export MK_CONFDIR=${MK_CONFDIR:-/etc/check_mk}
export MK_VARDIR=${MK_VARDIR:-/var/lib/check_mk_agent}
```
These paths can be configured via the launch daemon definition to a custom path without modifying the agent.

File: agents/check_mk_agent.macosx [hunk 2]
```diff
@@ -84,66 +249,296 @@ netstat -ntfinet | awk ' /^tcp/ { c[$6]++; } END { for (x in c) { print x, c[x];
...
+                Sensor="$(sed 's/^SM. //' <<<"$REPLY" | awk -F": " '{print $1}')"
+                echo "${Sensor}|enabled|disk-thermal|${Temp}000|50000|passive|60000|critical"
+                ;;
+            SMC*|SMB*)
+                Sensor="$(sed 's/^SM. //' <<<"$REPLY" | awk -F": " '{print $1}')"
+                echo "${Sensor}|enabled|thermal|${Temp}000|70000|passive|90000|critical"
+                ;;
+        esac
+    done
+elif type osx-cpu-temp >/dev/null
```
Comment (Notausstieg0309) [inline agents/check_mk_agent.macosx:281]: According to Linux Agent, this line should be changed by inserting and using the `inpath()` function as in the Linux agent:
```
# Function to replace "if type [somecmd]" idiom
# 'command -v' tends to be more robust vs 'which' and 'type' based tests
inpath() {
    command -v "${1:?No command to test}" >/dev/null 2>&1
}
```

**Review by Notausstieg0309 — COMMENTED** [review]
I would suggest some slight changes to get a bit closer to the linux agent implementation/behavior.

**Comment by mo-ki** [comment]
Hi catchdave,

I am sorry to close this PR, in particular because it took us so long.
I am not closing it, because it is awful, but because it is too large to asses it thoroughly. We currently don't have a bigger OSX environment to test it (and, judging from your comment, neither do you). Since I don't expect anyone to be confident enough to merge this in the next few years, I think closing it is the honest thing to do.
At the very least the change in `mk-job`  is worth a dedicated PR, I think.

Cheers,
Moritz

**Pull request closed without merging**